tools/image-registry.md

```diff
@@ -2,152 +2,155 @@
 
-* Have total control over where your images are being stored
-* Fully own your images' distribution pipeline
-* Ensure access control and security for your own Docker images
+### Prerequisites
 
-## Using Image Registry
+* [Kontena Cloud](https://cloud.kontena.io) account
+* [Kontena CLI](/tools/cli.md) with the `cloud` plugin. If you don't have the plugin installed, you can install (or upgrade) it with the: `kontena plugin install cloud` command.
 
-* [Create Image Registry Service](image-registry.md#create-image-registry-service)
-  * [Local Storage Backend](image-registry.md#local-storage-backend)
-  * [Amazon S3 Storage Backend](image-registry.md#amazon-s3-storage-backend)
-  * [Azure Storage Backend](image-registry.md#azure-storage-backend)
-* [Accessing Image Registry](image-registry.md#accessing-image-registry)
-* [TLS/SSL](image-registry.md#tlsssl)
-* [Authentication](image-registry.md#authentication)
+### Using with Docker CLI
 
-### Create Image Registry Service
+#### Login with Docker CLI
 
-#### Local Storage Backend
+#### Automated way
 
 ```
-$ kontena registry create
+$ kontena cloud image-repository docker-login
 ```
 
-#### Amazon S3 Storage Backend
+#### Manual way
 
-Write Amazon S3 access keys to Kontena Vault:
+* Create a permanent Kontena Cloud OAuth2 token:
 
-```
-$ kontena vault write REGISTRY_STORAGE_S3_ACCESSKEY <access_key>
-$ kontena vault write REGISTRY_STORAGE_S3_SECRETKEY <secret_key>
-```
+  ```
+  $ kontena cloud token create <NAME>
+  ```
 
-Create registry service:
+* Login with Docker CLI using your Kontena Cloud username and permanent token from the previous step:
 
-```
-$ kontena registry create --s3-bucket=<bucket_name> --s3-region=<optional_aws_region> --s3-v4auth
-```
+  ```
+  $ docker login -u <USERNAME> images.kontena.io
+  ```
 
-#### Azure storage backend
+#### Build & Push an image with Docker CLI
 
-Write Azure account key to Kontena Vault:
+Build:
 
 ```
-$ kontena vault write REGISTRY_STORAGE_AZURE_ACCOUNTKEY <azure_account_key>
+$ docker build -t images.kontena.io/<ORG>/<REPO>:<TAG> .
 ```
 
-Create registry service:
+Push:
 
 ```
-$ kontena registry create --azure-account-name=<account_name> --azure-container-name=<container_name>
+$ docker push images.kontena.io/<ORG>/<REPO>:<TAG>
 ```
 
-### Accessing Image Registry
+#### Pull an image with Docker CLI
 
-Before you can push images to the registry, you should set up the Kontena VPN service. In addition, you must set `--insecure-registry=registry.<grid_name>.kontena.local` in your local Docker daemon configuration.
+```
+$ docker pull images.kontena.io/<ORG>/<REPO>:<TAG>
+```
 
-Building and pushing an image to the registry:
+### List image repositories
 
 ```
-$ docker build -t registry.<grid_name>.kontena.local/myimage:mytag .
-$ docker push registry.<grid_name>.kontena.local/myimage:mytag
+$ kontena cloud image-repository list --org <ORG>
 ```
 
-Deploying an image from the registry:
+### Create a repository
```

Comment:
We don't have yet `create` command in the PR

Reply:
Should we have?

Reply:
Not necessary I think. The first push will create the repo and fire all needed events to track it. At least I don't see missing it as a blocker.

Reply:
IMHO we should add it just to make it logical for users. We can document that a repository can be created with the cli by just pushing a tag to new repo.

Reply:
create command is not needed imo - just push & forget